packages/calcite-components/src/components/sheet/sheet.scss

```diff
@@ -0,0 +1,222 @@
+  --calcite-sheet-max-width-internal: var(--calcite-sheet-max-width, 40vw);
+  --calcite-sheet-min-width-internal: var(--calcite-sheet-min-width, 40vw);
+}
+
+:host([position^="block"][height-scale="s"]) {
+  --calcite-sheet-min-height-internal: var(--calcite-sheet-min-height, 10vh);
+  --calcite-sheet-height-internal: var(--calcite-sheet-height, 20vh);
+  --calcite-sheet-max-height-internal: var(--calcite-sheet-max-height, 30vh);
+}
+
+:host([position^="block"][height-scale="m"]) {
+  --calcite-sheet-min-height-internal: var(--calcite-sheet-min-height, 10vh);
+  --calcite-sheet-height-internal: var(--calcite-sheet-height, 40vh);
+  --calcite-sheet-max-height-internal: var(--calcite-sheet-max-height, 50vh);
+}
+
+:host([position^="block"][height-scale="l"]) {
+  --calcite-sheet-min-height-internal: var(--calcite-sheet-min-height, 10vh);
+  --calcite-sheet-height-internal: var(--calcite-sheet-height, 60vh);
+  --calcite-sheet-max-height-internal: var(--calcite-sheet-max-height, 70vh);
+}
+
+.scrim {
+  --calcite-scrim-background: var(--calcite-sheet-scrim-background, var(--calcite-sheet-scrim-background-internal));
+  @apply fixed inset-0 flex overflow-y-hidden;
+}
+
+:host([opened]) {
+  @apply opacity-100;
+  visibility: visible !important;
+  transition-delay: 0ms;
+}
+
+.content {
+  @apply relative box-border block p-0 z-modal max-h-full max-w-full;
+  background-color: theme("colors.background.foreground.1");
+  max-block-size: 100%;
+  visibility: hidden;
+  transition: transform var(--calcite-internal-animation-timing-medium) $easing-function,
+    visibility 0ms linear var(--calcite-internal-animation-timing-medium),
+    opacity var(--calcite-internal-animation-timing-medium) $easing-function;
+  transform: var(--calcite-sheet-hidden-position-internal);
+}
+
+.container--open .content {
+  transform: translate3d(0, 0, 0);
+}
+
+:host([display-mode="float"]) .content,
+:host([display-mode="float"]) .container {
+  @apply rounded;
+}
+
+:host([display-mode="float"]) .container {
+  @apply m-3;
+}
+
+.container--open {
+  @apply opacity-100;
+  visibility: visible !important;
+  transition-delay: 0ms;
+  .content {
+    @apply pointer-events-auto visible opacity-100;
+    transition: transform var(--calcite-internal-animation-timing-medium) $easing-function, visibility 0ms linear,
+      opacity var(--calcite-internal-animation-timing-medium) $easing-function,
+      max-inline-size var(--calcite-internal-animation-timing-medium) $easing-function,
+      max-block-size var(--calcite-internal-animation-timing-medium) $easing-function;
+    transition-delay: 0ms;
+  }
+}
+
+:host([position="inline-start"]) .content,
+:host([position="inline-end"]) .content {
+  @apply h-full;
+}
+
+:host([position="block-start"]) .content,
+:host([position="block-end"]) .content {
+  @apply w-full;
+}
+
+/**
+ * Conditional styles for when Sheet is slotted in Shell
+ */
+
+.container.slotted-in-shell {
```

Comment:
Not critical for this PR, but is the compound class selector necessary to apply these styles or would `.slotted-in-shell` suffice?

Reply:
Copied from Modal. so refactor may be needed there as well.